<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>2nd Task</title>
</head>
<body>

    <script>
        class HttpError extends Error {
            constructor(response) {
                super(`${response.status} for ${response.url}`);
                this.name = 'HttpError';
                this.response = response;
            }
        }

        async function loadJson(url) {
            let response = await fetch(url);
            if(response.status == 200){
                return response.json();
            } else {
                throw new HttpError(response);
            }
        }

        async function demoGithubUser() {
            let name = prompt("Enter a name?", "iliakan");
            let response;
            try {
                response = await loadJson(`https://api.github.com/users/${name}`);
                alert(`Full name: ${response.name}.`);
            } catch(err){
                if (err instanceof HttpError && err.response.status == 404) {
                    alert("No such user, please reenter.");
                    return demoGithubUser();
                } else {
                    throw err;
                }
            }
        }

        demoGithubUser();
    </script>
    
</body>
</html>